hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/DecommissionUtils.java

```diff
@@ -126,7 +126,8 @@ public static Map<String, Object> getCountsMap(DatanodeDetails datanode, JsonNod
                                                  Map<String, Object> countsMap, String errMsg)
       throws IOException {
     for (int i = 1; i <= numDecomNodes; i++) {
-      if (datanode.getHostName().equals(counts.get("tag.datanode." + i).asText())) {
+      String dNHostName = (counts.get("tag.datanode." + i) != null) ? (counts.get("tag.datanode." + i).asText()) : "";
+      if (datanode.getHostName().equals(dNHostName)) {
```

Comment:
Not sure if we should implement it but can we Instead of silently defaulting to an empty string, it might be useful to log a warning message when a datanode tag is missing. This can help in debugging.

But even if we don't do it I am good with the changes.

```
    String datanodeTag = "tag.datanode." + i;
    JsonNode node = counts.get(datanodeTag);
    if (node == null) {
        LOG.warn("Datanode tag {} is missing from counts", datanodeTag);
        continue;
    }
    String datanodeHostName = node.asText();
    if (datanode.getHostName().equals(datanodeHostName)) {
        // Your existing logic here
    }
```

Reply:
I feel this is not needed as this information is not much useful to log and we can make it to empty string. And mostly this information is accessible over CLI also.